dev/app.js

```diff
@@ -5,7 +5,7 @@ var search = instantsearch({
   appId: 'latency',
   apiKey: '6be0576ff61c053d5f9a3225e2a90f76',
   indexName: 'instant_search',
-  urlSync: true
+  urlSync: window.hasHistoryAPI === false ? {useHash: true} : true
```

Comment:
Why not using a normal feature detection which does not rely on conditional comments for IE9 only.
E.g. Modernizr does it like that: https://github.com/Modernizr/Modernizr/blob/fc6e5f47a0df837d568007f165b5331acdf1e26a/feature-detects/history.js#L40
`(window.history && 'pushState' in window.history)`